### PR TITLE
add WormLikeChainOffsetPotential

### PR DESCRIPTION
**Pull request by yutakasi634**
WormLikeChain potential with the offset of the distance between two particles.
This PR contains
- Implementation
- test
- document (en, ja)
.
The offset version is separated from WormLikeChain potential to avoid calculation overhead in the original.

### CODE REVIEW

File: book/content.en/docs/reference/forcefields/local/bondlength/WormLikeChainOffsetPotential.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
++++
+title = "WormLikeChainOffset"
+weight = 600000
```
Comment (ToruNiina) [inline book/content.en/docs/reference/forcefields/local/bondlength/WormLikeChainOffsetPotential.md:3]: This value regulates the order in the menu bar. I think the best location of this potential is the next to the WormLikeChain potential, so `700000` is better.
```suggestion
weight = 700000
```

File: mjolnir/forcefield/local/WormLikeChainOffsetPotential.cpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#include <mjolnir/forcefield/local/WormLikeChainOffsetPotential.hpp>
+
+#ifndef MJOLNIR_SEPARATE_BUILD
+#error "MJOLNIR_SEPARATE_BUILD flag is required"
+#endif
+
+namespace mjolnir
+{
+    template class WormLikeChainOffsetPotential<double>;
+    template class WormLikeChainOffsetPotential<float>;
```
Comment (ToruNiina) [inline mjolnir/forcefield/local/WormLikeChainOffsetPotential.cpp:10]: Although it is a nitpicking, normally, we don't have any space here.
```suggestion
template class WormLikeChainOffsetPotential<double>;
template class WormLikeChainOffsetPotential<float>;
```

**Review by ToruNiina — CHANGES_REQUESTED** [review]
Basically it looks good to me.

But I have several very nitpicky comments (with suggestions). Could you resolve them?

**Comment by yutakasi634** [comment]
Thank you for your good suggestion.
I resolved those points!

**Review by ToruNiina — CHANGES_REQUESTED** [review]
Oh, wait, don't you have a test?

**Comment by yutakasi634** [comment]
I'm so sorry...
I added the test and fixed some parts.
Please check this.

**Review by ToruNiina — APPROVED** [review]
LGTM.

**Pull request merged**